code/datums/components/bakeable.dm

```diff
@@ -67,8 +67,9 @@
 	var/atom/original_object = parent
 	var/obj/item/plate/oven_tray/used_tray = original_object.loc
 	var/atom/baked_result = new bake_result(used_tray)
-	baked_result.reagents.clear_reagents()
-	original_object.reagents?.trans_to(baked_result, original_object.reagents.total_volume)
+	if(baked_result.reagents) //make space and tranfer reagents if it has any
+		baked_result.reagents.clear_reagents()
+		original_object.reagents?.trans_to(baked_result, original_object.reagents.total_volume)
```

Comment:
```suggestion
		original_object.reagents.trans_to(baked_result, original_object.reagents.total_volume)
```

Reply:
I'm not sure that's safe since we don't do a null check on it's reagents but ok